docs/explanations/decisions/0007-subpackage-structure.md

```diff
@@ -0,0 +1,95 @@
+# 7. Sub-package Structure
+
+Date: 2024-04-22
+
+## Status
+
+Accepted
+
+## Context
+
+[](./0004-repository-structure) proposed a top level repository structure. This builds upon it, suggesting a top level structure divided into:
+
+- `ophyd_async.core`: Core classes like `Device`, `Signal` and `AsyncStatus`
+- `ophyd_async.epics`: Epics specific signals and devices
+- `ophyd_async.tango`: Tango specific signals and devices
+- `ophyd_async.fastcs`: FastCS (EPICS or Tango) devices like PandA
+- `ophyd_async.planstubs`: Plan stubs for various flyscan functionality
+- `ophyd_async.sim`: Simulated devices for demos and tests
+
+This ADR proposes a public sub-package structure. The internal private structure should be flat, but can change according to the number of classes in a public package.
+
+## Decision
+
+### core
+
+There will be a flat public namespace under core, with contents reimported from an underscore prefixed python files, e.g.:
+
+- `_status.py` for `AsyncStatus`, `WatchableAsyncStatus`, etc.
+- `_protocol.py` for `AsyncReadable`, `AsyncStatus`, etc.
```

Comment:
`AsyncStatus` is from `_status.py` and not `_protocol.py`.

Reply:
Ah there is a bit of a circular thing going on here.

Reply:
```suggestion
- `_protocol.py` for `AsyncReadable`, `AsyncStageable`, etc.
```